pep-0682.rst

```diff
@@ -0,0 +1,183 @@
+PEP: 682
+Title: Format Specifier for Signed Zero
+Author: John Belmonte <john@neggie.net>
+Sponsor: Mark Dickinson <dickinsm@gmail.com>
+Discussions-To:
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 29-Jan-2022
+Python-Version: 3.11
+Post-History:
+
+
+Abstract
+========
+
+Though ``float`` and ``Decimal`` types can represent `signed zero`_, in many 
+fields of mathematics, negative zero is surprising or unwanted -- especially
+in the context of displaying an (often rounded) numerical result.  This PEP
+proposes an extension to the `string format specification`_, allowing negative
+zero to be normalized to positive zero.
+
+.. _`signed zero`: https://en.wikipedia.org/wiki/Signed_zero
+.. _`string format specification`: https://docs.python.org/3/library/string.html#formatstrings
+
+
+Motivation
+==========
+
+Here is negative zero:
+
+.. code-block:: pycon
+
+    >>> x = -0.
+    >>> x
+    -0.0
+
+When formatting a number, negative zero can result from rounding.  Assuming
+the user's intention is truly to discard precision, the distinction between
+negative and positive zero of the rounded result might be considered an
+unwanted artifact:
+
+.. code-block:: pycon
+
+    >>> for x in (.002, -.001, .060):
+    ...     print(f'{x: .1f}')
+     0.0
+    -0.0
+     0.1
+
+There are various approaches to clearing the sign of a negative zero.  It
+can be achieved without a conditional by adding positive zero:
+
+.. code-block:: pycon
+
+    >>> x = -0.
+    >>> x + 0.
+    0.0
+
+To normalize negative zero when formatting, it is necessary to perform
+a redundant (and error-prone) pre-rounding of the input:
+
+.. code-block:: pycon
+
+    >>> for x in (.002, -.001, .060):
+    ...     print(f'{round(x, 1) + 0.: .1f}')
+     0.0
+     0.0
+     0.1
+
+What we would like instead is a first-class option to normalize negative
+zero, on top of everything else that numerical string formatting already
+offers.
```

Comment:
Should we be providing evidence that the feature is needed / frequently requested here, or do you think that can be saved for the PEP discussion? It's a reasonably frequently-asked question on Stack Overflow, for example. Some samples are below; there are plenty more where those came from:

One answer (to a question specific to printing negative zeros) showing the use of a post-formatting regex: https://stackoverflow.com/a/36604981/270986

Another question that's more clearly about handling negative zeros resulting from the rounding that's implicit in float formatting:
https://stackoverflow.com/questions/41564311/ironpython-formatting-issue-with-modulo-operator-negative-zero

Another one, with me proposing a non-solution in comments:
https://stackoverflow.com/questions/66283651/how-to-handle-rounding-to-negative-zero-in-python-docstring-tests

A Java variant (accepted solution also involves a post-formatting regex replacement):
https://stackoverflow.com/questions/11929096/negative-sign-in-case-of-zero-in-java

And another one, for Objective C:
https://stackoverflow.com/questions/10969399/prevent-small-negative-numbers-printing-as-0

Reply:
added most of these (and How To Teach This) -- thank you!